src/Native/Unix/Common/pal_compiler.h

```diff
@@ -13,9 +13,13 @@
 #define END_EXTERN_C
 #endif
 
+#ifndef __has_extension
+#define __has_extension(...) 0
+#endif
+
 #ifdef static_assert
 #define c_static_assert(e) static_assert((e),"")
-#elif defined(__has_extension) && __has_extension(c_static_assert)
+#elif __has_extension(c_static_assert)
```

Comment:
Is it change needed if we have `#define __has_extension()` above? We can probably define these in makefile to keep the the files closer to the source.

Reply:
It's much simpler to keep the check in this header rather than spread it around in a Makefile. Especially since https://github.com/mono/corefx/pull/46/files#diff-a4c8175f5b95c7ffaebf40eccc196d7fR16 will be a no-op if it's already defined. Also, this change will be upstreamed.